compiler/code-gen/declarations.cpp

```diff
@@ -936,7 +932,7 @@ void ClassDeclaration::compile_accept_visitor_methods(CodeGenerator &W, ClassPtr
     compile_accept_visitor(W, klass, "CommonMemoryEstimateVisitor");
   }
 
-  if (klass->need_instance_cache_visitors) {
+  if (klass->need_instance_cache_visitors && !G->is_output_mode_k2()) {
```

Comment:
Same for previous if with `need_instance_memory_estimate_visitor`

Reply:
+, thx!